### PR TITLE
Phil spel highline pde

### LINKED ISSUE #123: Add exercise that models slackline as a PDE
This can be an extension of the slackline as linear system and #56

### PR DESCRIPTION
**Pull request by PhiSpel**
- [ ] Should finite differences be introduced elsewhere?
- [x] Does Gerd agree with this way of including the exercise?

closes #123

### CODE REVIEW

File: content/11_pde/01_modellierung_einer_highline_dynamisch_pde.ipynb [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    "\n",
+    "Wie in [](content:references:highline_dynamic) wollen wir nun die Highline aus [](content:references:highline_static) als partielle Differentialgleichung (Partial Differential Equation, PDE) lösen."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f403e856",
+   "metadata": {},
+   "source": [
+    "Auch wenn es nicht so aussah, beinhaltet bereits die die Kräftebilanzierung aus [](content:references:highline_static) eine partielle Differentialgleichung. Für einen allgemeinen Massepunkt mit den Koordinaten $\\mathbf{z}_i = \\begin{bmatrix} x_i(t) \\\\ y_i(t) \\end{bmatrix}$ ergibt sich aus der Kräftebilanz $m_i \\ddot{\\mathbf{z}}_i = \\mathbf{F}_{i,\\text{Feder}} + \\mathbf{F}_{i,\\text{Dämpfer}} + \\mathbf{F}_{i,\\text{ext}}, i = 1,...,N$ mit $\\mathbf{F}_{i,\\text{Feder}} = k \\frac{\\partial^2 \\mathbf{z}}{\\partial x^2}$, $\\mathbf{F}_{i,\\text{Dämpfer}} = d \\frac{\\partial^2}{\\partial x^2} \\frac{\\partial \\mathbf{z}}{\\partial t}$ und $m_i = m, k_i = k, d_i = d$ die PDE\n",
```
Comment (sandro-elsweijer) [inline content/11_pde/01_modellierung_einer_highline_dynamisch_pde.ipynb:19]: The formatting of these equations seems weird.

File: content/11_pde/01_modellierung_einer_highline_dynamisch_pde.ipynb [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    "\n",
+    "oder\n",
+    "\n",
+    "$$m \\mathbf{z}_{tt} = k \\mathbf{z}_{xx} + d {\\mathbf{z}_t}_{xx} + \\mathbf{F}_{\\text{ext}}(t).$$\n",
+    "\n",
+    "```{admonition} Hinweis \n",
+    "Die Ableitung nach dem Ort in beide Raumrichtungen $\\frac{\\partial}{\\partial x}$ ist nicht zu verwechseln mit der einzelnen Raumkoordinaten $x_i$.\n",
+    "```\n",
+    "\n",
+    "Um diese PDE in ein System von ODEs zu überführen, muss ein sogenannter Differenzenstern für die Ortsableitungen gebildet werden. Im Eindimensionalen ist der Differenzenstern sehr einfach als zentrale Differenzenquotienten, mit\n",
```
Comment (sandro-elsweijer) [inline content/11_pde/01_modellierung_einer_highline_dynamisch_pde.ipynb:31]: The last sentence is hard to read.

File: content/11_pde/01_modellierung_einer_highline_dynamisch_pde.ipynb [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    "\n",
+    "$$\\frac{\\partial^2 z}{\\partial x^2} \\approx \\frac{z(x-h)-2z(x)+z(x+h)}{h^2} \\to \\frac{\\partial z_i}{\\partial x^2} = \\frac{z_{i-1}-2z_i+z_{i+1}}{dx^2}.$$\n",
```
Comment (sandro-elsweijer) [inline content/11_pde/01_modellierung_einer_highline_dynamisch_pde.ipynb:33]: I would start for the difference quotient for the first derivative and build the difference quotient for the second derivative from this. This way it gets clear where the 2z(x) comes from.

File: content/11_pde/01_modellierung_einer_highline_dynamisch_pde.ipynb [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    "\n",
+    "Nach der Übertragung von $k$ und $d$ auf Elementweise $k$ und $d$ entsteht die bekannte Gleichung aus [](content:references:highline_static):\n",
+    "\n",
+    "$$m \\ddot{\\mathbf{z}}_i = k(\\mathbf{z}_{i-1}-\\mathbf{z}_{i}) + d(\\dot{\\mathbf{z}}_{i-1} - \\dot{\\mathbf{z}}_{i}) - k(\\mathbf{z}_{i}-\\mathbf{z}_{i+1}) - d(\\dot{\\mathbf{z}}_{i} -\\dot{\\mathbf{z}}_{i+1}) + \\mathbf{F}_{i,\\text{ext}}$$\n",
+    "\n",
+    "Achtung: Für den ersten und den letzten Massenpunkt ($i=1$ und $i=N$) sind zusätzlich die Geschwindigkeit und Verschiebung der Massenpunkte an den Rändern für die finiten Differenzen notwendig. Es entsteht also ein System mit $N+4$ Unbekannten (beziehungsweise $2N$ Unbekannte, falls wir die Vektoren in ihre Raumrichtungen zerlegen) und nur $N$ Gleichungen entsteht.\n",
```
Comment (sandro-elsweijer) [inline content/11_pde/01_modellierung_einer_highline_dynamisch_pde.ipynb:39]: One "entsteht" too much

File: content/11_pde/01_modellierung_einer_highline_dynamisch_pde.ipynb [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    "\n",
+    "Daher müssen Randbedingungen festgelegt werden. Diese ist für die Highline ganz logisch und einfach festzulegen: Die  Geschwindigkeit und Verschiebung der äußeren Elemente sind natürlich $\\mathbf{\\dot{z}}_0=\\mathbf{\\dot{z}}_{N+1}=0, \\mathbf{z}_0=\\mathbf{z}_{N+1}=0$. Damit werden die Gleichungen für $i=1$ und $i=N$ etwas kürzer und es ergeben sich die beiden zusätzlichen Gleichungen für $\\ddot{z}$ bei $i=0$ und $i=N+1$ und wir erhalten das Gleichungssystem\n",
```
Comment (sandro-elsweijer) [inline content/11_pde/01_modellierung_einer_highline_dynamisch_pde.ipynb:41]: Maybe we could insert a comment that this type of boundary condition is a Dirichlet boundary condition and that there is also a different type of BC, the Neumann BC, which sets the spacial derivative for this knot.

File: content/11_pde/01_modellierung_einer_highline_dynamisch_pde.ipynb [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    " m \\ddot{\\mathbf{z}}_i &= k(\\mathbf{z}_{i-1}-\\mathbf{z}_{i}) + d(\\dot{\\mathbf{z}}_{i-1} - \\dot{\\mathbf{z}}_{i}) - k(\\mathbf{z}_{i}-\\mathbf{z}_{i+1}) - d(\\dot{\\mathbf{z}}_{i} -\\dot{\\mathbf{z}}_{i+1}) + \\mathbf{F}_{i,\\text{ext}}, \\hskip4em \\text{für } i=2,...,N-1 \\notag \\\\\n",
+    " m \\ddot{\\mathbf{z}}_N &= -k \\mathbf{z}_N - d \\dot{\\mathbf{z}}_N + k(\\mathbf{z}_{N-1}-\\mathbf{z}_N) + d(\\dot{\\mathbf{z}}_{N-1}-\\dot{\\mathbf{z}}_N) + \\mathbf{F}_{N,\\text{ext}} \\notag \\\\\n",
+    " m \\ddot{\\mathbf{z}}_{N+1} &= 0 \\notag\n",
+    "\\end{align}\n",
+    "$$\n",
+    "\n",
+    ", für das wir die Lösung bereits gelernt haben.\n",
+    "\n",
+    "```{admonition} Hinweis \n",
+    "Diese Lösungsstrategie nennt sich übrigens **Finite-Differenzen-Verfahren** (FDM):\n",
```
Comment (sandro-elsweijer) [inline content/11_pde/01_modellierung_einer_highline_dynamisch_pde.ipynb:56]: Better Finite-Differenzen-Methode, if you want to use the acronym FDM

File: content/11_pde/01_modellierung_einer_highline_dynamisch_pde.ipynb [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    "\n",
+    "Die Ableitungen in der DGL werden durch finite Differenzenquotienten ersetzt. Zusammen mit den Randbedingungen ergibt sich ein lineares (oder auch nichtlineares) Gleichungssystem.\n",
```
Comment (sandro-elsweijer) [inline content/11_pde/01_modellierung_einer_highline_dynamisch_pde.ipynb:58]: PDE not DGL, and only the spacial derivatives get exchanged. After that you have a system of ODEs and not a linear or non-linear system of algebraic equations.

**Review by sandro-elsweijer — CHANGES_REQUESTED** [review]
This should be the introduction to PDEs and because of that you should mention the difference between ODEs and PDEs at least in one sentence. Apart from that, this looks great.

File: content/11_pde/01_modellierung_einer_highline_dynamisch_pde.ipynb (hunk 7 above)
Comment (PhiSpel) [reply content/11_pde/01_modellierung_einer_highline_dynamisch_pde.ipynb]: see 42db10e

File: content/11_pde/01_modellierung_einer_highline_dynamisch_pde.ipynb (hunk 6 above)
Comment (PhiSpel) [reply content/11_pde/01_modellierung_einer_highline_dynamisch_pde.ipynb]: see 42db10e

File: content/11_pde/01_modellierung_einer_highline_dynamisch_pde.ipynb (hunk 5 above)
Comment (PhiSpel) [reply content/11_pde/01_modellierung_einer_highline_dynamisch_pde.ipynb]: see 42db10e

File: content/11_pde/01_modellierung_einer_highline_dynamisch_pde.ipynb (hunk 4 above)
Comment (PhiSpel) [reply content/11_pde/01_modellierung_einer_highline_dynamisch_pde.ipynb]: see 42db10e

File: content/11_pde/01_modellierung_einer_highline_dynamisch_pde.ipynb (hunk 3 above)
Comment (PhiSpel) [reply content/11_pde/01_modellierung_einer_highline_dynamisch_pde.ipynb]: see 42db10e

File: content/11_pde/01_modellierung_einer_highline_dynamisch_pde.ipynb (hunk 2 above)
Comment (PhiSpel) [reply content/11_pde/01_modellierung_einer_highline_dynamisch_pde.ipynb]: see 42db10e

File: content/11_pde/01_modellierung_einer_highline_dynamisch_pde.ipynb (hunk 1 above)
Comment (PhiSpel) [reply content/11_pde/01_modellierung_einer_highline_dynamisch_pde.ipynb]: I wanted to introduce both the parial/partial x convention as well as the z_x convention. I changed the equations in the text to z_x convention, see 42db10e

**Comment by PhiSpel** [comment]
> This should be the introduction to PDEs and because of that you should mention the difference between ODEs and PDEs at least in one sentence. Apart from that, this looks great.

Thank you for the helpful comments! Now, everything should be in order

**Review by sandro-elsweijer — APPROVED** [review]

**Pull request merged**